### PR TITLE
Add some logging docs

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "better docs".

https://github.com/esl/MongooseIM/blob/logging-docs/doc/developers-guide/logging.md

### CODE REVIEW

File: doc/developers-guide/logging.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+-include("mongoose_logger.hrl").
+```
+
+or
+
+```erlang
+-include("mongoose.hrl").
+```
+
+# Logging macroses
```
Comment (arkgil) [inline doc/developers-guide/logging.md]: Should be "macros".

File: doc/developers-guide/logging.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                       "reason=~p:~p user=~ts stacktrace=~1000p",
+                       [Class, Reason, LUser, Stacktrace]),
+            erlang:raise(Class, Reason, Stacktrace)
+    end
+```
+
+`user=~ts` is often used too.
+
+
+A common way to name event is `event=function_name_failed`.
```
Comment (arkgil) [inline doc/developers-guide/logging.md]: Let's not advise it. Naming things is hard and our functions rarely have good names :P We can check which function has failed by enabling lager's `function` metadata.

**Comment by arkgil** [comment]
What I don't like about it is that `event` and `reason` fields won't be caught by backends like https://github.com/esl/lager_graylog as a metadata, and thus won't interpreted by the logging aggregator service, Graylog in this case, as a separate field. Sure, you can always add a parser to the log aggregator so that it extracts those fields from the log message, but still, this is less then ideal. Maybe we could set the `event` and `reason` as a metadata? And then configure lager to include that metadata in the message.

File: doc/developers-guide/logging.md (hunk 2 above)
Comment (arcusfelis) [reply doc/developers-guide/logging.md]: > A common way to name event is `event=function_name_failed`, but use common sense :)

**Comment by arcusfelis** [comment]
@arkgil we were thinking about it, but nobody actually managed to make a code we can merge :). Actually, splunk parses KEY=VALUE just fine. ELK too after defining parser. It's also good to have a json lager backend for ELK.

**Comment by arcusfelis** [comment]
@arkgil last trial I remember https://github.com/esl/MongooseIM/pull/1255/files

**Comment by arkgil** [comment]
@arcusfelis I think that our logging standards require deeper discussion. I believe this PR is a good start! 🌮

**Comment by arcusfelis** [comment]
@arkgil sure :) I usually write the same things in comments, so just put everything together.

File: doc/developers-guide/logging.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+If an user sets log level 5, than he would see all messages in logs.
+
+Levels 3 or 2 are usually used for production systems.
+
+
+# Logging format
+
+We use modified [logfmt](https://brandur.org/logfmt) format.
+
+This format is Splunk and ELK friendly.
```
Comment (ludwikbukowski) [inline doc/developers-guide/logging.md]: [Splunk](https://www.splunk.com/en_us/solutions/solution-areas/log-management.html) and [ELK](https://www.elastic.co/elk-stack)

File: doc/developers-guide/logging.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+WARNING_MSG("event=warning_event reason=~1000p", [Arg]),
+```
+
+Use them to log on corresponding log level.
+
+Don't use `ERROR_MSG` for cases that are not errors.
+
+# Logging levels
+
+A system operator can choose log level by setting loglevel in `mongooseim.cfg`.
```
Comment (ludwikbukowski) [inline doc/developers-guide/logging.md]: where is this file?

File: doc/developers-guide/logging.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+%% Use
+?ERROR_MSG("event=check_password_failed reason=~p", [Reason])
+
+%% Don't use
+?ERROR_MSG("event=check_password_failed, reason=~p", [Reason])
+```
+
+# Filtering logs by module
+
+Setting loglevel in ejabberd.cfg to debug leads to a flood of messages in logs.
```
Comment (ludwikbukowski) [inline doc/developers-guide/logging.md]: What do you think?
```
Setting loglevel to `debug` leads to a flood of messages in logs.
To set different loglevel for just one module, call:
```

File: doc/developers-guide/logging.md (hunk 4 above)
Comment (arcusfelis) [reply doc/developers-guide/logging.md]: still? :trollface:

File: doc/developers-guide/logging.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+`user=~ts` is often used too.
+
+
+A common way to name an error event is `event=function_name_failed`.
+For example, `event=remove_user_failed`. Use the advice critically, it would
+not work well for any function. Counter example:
+
+```erlang
+handle_info(Info, State) ->
+    ?ERROR_MSG("issue=unexpected_info_received info=~1000p", [Info]),
```
Comment (fenek) [inline doc/developers-guide/logging.md]: I disagree, `event=unexpected_info_received` sounds equally well IMO.

File: doc/developers-guide/logging.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    {noreply, State}.
+```
+
+Log messages should not contain new lines.
+
+We usually use `~1000p` to log long datastructures.
+
+Use whitespace as a field separator:
```
Comment (fenek) [inline doc/developers-guide/logging.md]: What should be done about values that contain whitespaces?

File: doc/developers-guide/logging.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+# Logging format
+
+We use modified [logfmt](https://brandur.org/logfmt) format.
+
+This format is [Splunk](https://www.splunk.com/en_us/solutions/solution-areas/log-management.html)
+and [ELK](https://www.elastic.co/elk-stack) friendly.
+
+`event=something_interesting` field is required.
+
+`reason=~1000p` field is commonly used.
```
Comment (fenek) [inline doc/developers-guide/logging.md]: I think we could use an info about default message length limit enforced by lager.

**Pull request merged**